python/lsst/analysis/tools/tasks/associatedSourcesTractAnalysis.py

```diff
@@ -229,63 +218,33 @@ def applyAstrometricCorrections(self, dataJoined, astrometricCorrectionCatalogs,
-            with Matcher(meanRAs, meanDecs) as m:
-                idx, i1, i2, d = m.query_radius(
-                    bandPMs[self.config.astrometricCorrectionParameters["ra"]],
-                    bandPMs[self.config.astrometricCorrectionParameters["dec"]],
-                    (self.config.matchingRadius * u.mas).to(u.degree),
-                    return_indices=True,
-                )
-
-            catRAs = np.zeros_like(meanRAs)
-            catDecs = np.zeros_like(meanRAs)
-            pmRAs = np.zeros_like(meanRAs)
-            pmDecs = np.zeros_like(meanRAs)
-            parallaxes = np.zeros(len(meanRAs))
-            catRAs[i1] = bandPMs[self.config.astrometricCorrectionParameters["ra"]][i2]
-            catDecs[i1] = bandPMs[self.config.astrometricCorrectionParameters["dec"]][i2]
-            pmRAs[i1] = bandPMs[self.config.astrometricCorrectionParameters["pmRA"]][i2]
-            pmDecs[i1] = bandPMs[self.config.astrometricCorrectionParameters["pmDec"]][i2]
-            parallaxes[i1] = bandPMs[self.config.astrometricCorrectionParameters["parallax"]][i2]
-
-            pmDf = Table(
-                {
-                    "ra": catRAs * u.degree,
-                    "dec": catDecs * u.degree,
-                    "pmRA": pmRAs * u.mas / u.yr,
-                    "pmDec": pmDecs * u.mas / u.yr,
-                    "parallax": parallaxes * u.mas,
-                    "obj_index": meanRAs.index,
-                }
-            )
 
-            dataWithPM = join(bandSources, pmDf, keys="obj_index", join_type="left")
+        # Get the stellar motion catalog into the right format:
+        for key, value in self.config.astrometricCorrectionParameters.items():
+            astrometricCorrectionCatalog.rename_column(value, key)
+        astrometricCorrectionCatalog["ra"] *= u.degree
+        astrometricCorrectionCatalog["dec"] *= u.degree
+        astrometricCorrectionCatalog["pmRA"] *= u.mas / u.yr
+        astrometricCorrectionCatalog["pmDec"] *= u.mas / u.yr
+        astrometricCorrectionCatalog["parallax"] *= u.mas
+
+        dataWithPM = join(
+            dataJoined,
+            astrometricCorrectionCatalog,
+            keys="isolated_star_id",
+            join_type="left",
+            keep_order=True,
+        )
 
-            visits = bandSourcesDf["visit"].unique()
-            mjds = [visitTable.loc[visit]["expMidptMJD"] for visit in visits]
-            mjdTable = Table(
-                [astropy.time.Time(mjds, format="mjd", scale="tai"), visits], names=["MJD", "visit"]
-            )
-            dataWithMJD = join(dataWithPM, mjdTable, keys="visit", join_type="left")
-            # After astropy 7.0, it should be possible to use "keep_order=True"
-            # in the join and avoid sorting.
-            dataWithMJD.sort("__index__")
-            medianMJD = astropy.time.Time(np.median(mjds), format="mjd", scale="tai")
+        mjds = visitTable.loc[dataWithPM["visit"]]["expMidptMJD"]
```

Comment:
Probably outside the scope of this ticket but can we one day get the visitTable to be astropy and remove the uses of pandas.

Reply:
I would do it if I had finished 12 hours earlier...I think we can get rid of the rest of the pandas pretty easily on another ticket now though.